### PR TITLE
Loadingspinner

### PR DESCRIPTION
**Pull request by realswat1**
I implemented our loading spinner  by creating a component for it and delaying our video fetching by 2 sec , i will add an extra 8sec so that the test is very noticeable  
https://www.loom.com/share/cc22fff6691543e78a4c33514eb65287?sid=247f38ca-d5f4-4f1c-bfd6-99b30084814d

In other to test our loading spin in a an edge case of very poor internet connection we used  
function waitforme(millisec) {
    return new Promise(resolve => {
        setTimeout(() => { resolve('') }, millisec);
    })
}
app.get("/videos", async (req, res) => {
  console.log('check');
  try {
await waitforme(5000);
// logic of endpoint
} 
in our server file to go past async to create the artificial delay.
https://www.loom.com/share/ab857a1e2d464ebdb17640d4cc61e7b6?sid=f18942da-3fd1-4226-a0c7-85e6e6374056

### CODE REVIEW

File: exposee-ui/src/components/LoadingSpinner/LoadingSpinner.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    border-radius: 50%;
+    background-color: #3498db; /* Change the color of the dots if needed */
+    animation: bounce 0.5s infinite alternate;
+  }
+
+  .dot:nth-child(2) {
+    animation-delay: 0.15s;
+  }
+
+  .dot:nth-child(3) {
```
Comment (alfonsocheng) [inline exposee-ui/src/components/LoadingSpinner/LoadingSpinner.css]: haha this is a really impressive way to make the loading spinner!!! 
In most cases, it would make sense to just use a gif for spinner, and then in react show the spinner by default until the response comes back, then you show the actual content.

File: exposee-ui/src/components/Mainpage/Main.jsx [hunk 2]
```diff
@@ -21,6 +25,8 @@ function Main() {
         }
       } catch (error) {
         console.log("Error fetching videos", error);
+      } finally {
+        setLoading(false); // Set loading to false after videos are fetched
```
Comment (alfonsocheng) [inline exposee-ui/src/components/Mainpage/Main.jsx:29]: good idea to put a finally here!

**Review by alfonsocheng — APPROVED** [review]

**Comment by alfonsocheng** [comment]
Nice job, thanks for adding the test with the setTimeout!!
